vowpalwabbit/csoaa.cc

```diff
@@ -470,6 +470,23 @@ void do_actual_learning(ldf& data, base_learner& base)
   if (start_K > 0)
     for (size_t k=1; k<K; k++)
       LabelDict::del_example_namespaces_from_example(*data.ec_seq[k], *data.ec_seq[0], (data.all->audit || data.all->hash_inv));
+
```

Comment:
I'm confused about the application to csoaa.cc.   Why does this make sense?  Predicting the expected cost given features does not have a probability semantics in general.  Why not just go through the oaa.cc code path?  This seems like to doing the same thing in two places.